### PR TITLE
feat: add base chain & support for wallets

### PR DESCRIPTION
**Pull request by ice-chillios**
Added support for:
- Coinbase
- Keystore
- Exodus
- Keepkey
- Xdefi
- Walletconnect
- Okx
- All EVM Browser extensions

Missing and will add in other PR (as needs some polishes around that)
- Ledger
- Trezor
- Talisman (?)

### CODE REVIEW

**Review by anthon-dev — APPROVED** [review]
LGTM

File: packages/swapkit/helpers/src/helpers/asset.ts [hunk 1]
```diff
@@ -190,24 +193,29 @@ export const getAssetType = ({ chain, symbol }: { chain: Chain; symbol: string }
   if (symbol.includes("/")) return "Synth";
 
   switch (chain) {
+    case Chain.Arbitrum:
+      return [Chain.Ethereum, Chain.Arbitrum].includes(symbol as Chain) ? "Native" : "ARBITRUM";
```
Comment (towanTG) [inline packages/swapkit/helpers/src/helpers/asset.ts]: ARB is just an ARC-20 no?

File: packages/swapkit/helpers/src/helpers/asset.ts [hunk 2]
```diff
@@ -190,24 +193,29 @@ export const getAssetType = ({ chain, symbol }: { chain: Chain; symbol: string }
...
+    case Chain.Optimism:
+      return [Chain.Ethereum, Chain.Optimism].includes(symbol as Chain) ? "Native" : "OPTIMISM";
+    case Chain.Base:
+      return [Chain.Ethereum, Chain.Base].includes(symbol as Chain) ? "Native" : "BASE";
```
Comment (towanTG) [inline packages/swapkit/helpers/src/helpers/asset.ts]: What about base?

File: packages/swapkit/helpers/src/helpers/asset.ts (hunk 1 above)
Comment (ice-chillios) [reply packages/swapkit/helpers/src/helpers/asset.ts]: No xd ARC-20 is sth completely different than ERC20 - we had that convo around AVAX no?😅

File: packages/swapkit/helpers/src/helpers/asset.ts [hunk 3]
```diff
@@ -190,24 +193,29 @@ export const getAssetType = ({ chain, symbol }: { chain: Chain; symbol: string }
...
     case Chain.Cosmos:
       return symbol === "ATOM" ? "Native" : Chain.Cosmos;
-    case Chain.Kujira:
-      return symbol === Chain.Kujira ? "Native" : Chain.Kujira;
     case Chain.BinanceSmartChain:
       return symbol === "BNB" ? "Native" : "BEP20";
-    case Chain.Ethereum:
-      return symbol === Chain.Ethereum ? "Native" : "ERC20";
+
+    case Chain.Kujira:
```
Comment (towanTG) [inline packages/swapkit/helpers/src/helpers/asset.ts]: You could probably combine some of those cases and return
return symbol === chain ? "Native" : chain;

File: packages/swapkit/helpers/src/helpers/asset.ts (hunk 2 above)
Comment (towanTG) [reply packages/swapkit/helpers/src/helpers/asset.ts]: I mean: Is base anything more than a token on the chain?

File: packages/swapkit/helpers/src/helpers/asset.ts (hunk 1 above)
Comment (towanTG) [reply packages/swapkit/helpers/src/helpers/asset.ts]: I would classify ARB.ARB as an ARC-20 token

File: packages/swapkit/helpers/src/helpers/asset.ts (hunk 3 above)
Comment (ice-chillios) [reply packages/swapkit/helpers/src/helpers/asset.ts]: For some of them - sure - I will make "default" case like that

File: packages/swapkit/helpers/src/helpers/asset.ts (hunk 1 above)
Comment (towanTG) [reply packages/swapkit/helpers/src/helpers/asset.ts]: https://docs.arbitrum.foundation/concepts/arb-token

File: packages/swapkit/helpers/src/helpers/asset.ts (hunk 3 above)
Comment (towanTG) [reply packages/swapkit/helpers/src/helpers/asset.ts]: thanks

File: packages/swapkit/helpers/src/helpers/asset.ts (hunk 1 above)
Comment (ice-chillios) [reply packages/swapkit/helpers/src/helpers/asset.ts]: Skipping that classify as spoken - we will only show `Native` for gas assets to not confuse it with governance tokens

**Pull request merged**